WalletWasabi.Gui/Program.cs

```diff
@@ -151,10 +148,12 @@ private static async Task DisposeAsync()
 			AppDomain.CurrentDomain.UnhandledException -= CurrentDomain_UnhandledException;
 			TaskScheduler.UnobservedTaskException -= TaskScheduler_UnobservedTaskException;
 
-			if (disposeGui)
+			if (mainViewModel is { })
```

Comment:
For this, it would suffice to have `runGui` as parameter of this method.

Reply:
It is also called from AppMainAsync where the runGui parameter is missing. It could be solved only if I add a static bool to program.cs. IMO it is better if the terminate method figures out what to dispose by itself. Maybe late we have to call it from different points of the code where runGui is more complex to read.